doc/modules/linear_model.rst

```diff
@@ -735,7 +735,10 @@ of a single trial are modeled using a
 Logistic regression is implemented in :class:`LogisticRegression`. 
 This implementation can fit binary, One-vs-Rest, or multinomial logistic 
 regression with optional :math:`\ell_1`, :math:`\ell_2` or Elastic-Net 
-regularization. Note that regularization is applied by default.
+regularization.
+
+**Note that regularization is applied by default.** No regularization amounts
+to setting C to a very high value.
```

Comment:
Maybe mention 
"""
Setting C to a very high value amounts to no regularisation together with better numerical stability.
"""

Reply:
also mention ``regularization='none'`` maybe? Is it mentioned somewhere which solvers support that?

Reply:
@rth @jnothman honestly I'm not sure I understand that statement.
What are we claiming is more numerical stable than what?

Reply:
@amueller I was just paraphrasing your initial description in https://github.com/scikit-learn/scikit-learn/issues/14070#issue-454846266 

That even if theortically we don't want to regularize, in practice having some small regularisation is a good thing for the numerical stability.

My formulation can certainly be improved.